### DISCUSSION TITLE
:grey_question: Challenge around 'string_to_array` :thinking:

### DISCUSSION THREAD

**Discussion by adriens** [duckdb/duckdb#6674] [Ideas]
# :grey_question: Context

I'm going on further on DuckDB explorations on a project (I'll blog about it)... and I'm not facing a common pattern I can fix with Python... but I wonder if I could do this in pure `SQL` as part of ETL part of the project.

# :information_source: Problem pattern

- :heavy_check_mark: I have a table with a `VARCHAR` column (let's call the column `keywords`)
- :heavy_check_mark: I split the column (`string_to_array(trim(keywords), ', ')`) into an array and put the output array into a new dedicated column
- :grey_question:  **How to ut all the keyword into a new single column table (one row per keyword instance) :thinking:** 

:point_up: I can do this in Python by fetching rows... but I have the feeling that this common problem could be achieved in pure `SQL` : am I wrong ?

**Comment by Mytherin** [comment]
You're probably looking for `UNNEST`:

```sql
SELECT unnest(string_split('kw1 kw2 kw3', ' ')) as keywords;
┌──────────┐
│ keywords │
│ varchar  │
├──────────┤
│ kw1      │
│ kw2      │
│ kw3      │
└──────────┘
```

**Reply by adriens** [reply]
:open_mouth: Let me give a try and tell you... but it looks like what I'm really looking for :heavy_heart_exclamation:

**Reply by adriens** [reply]
:pray: IT WORKS awesomely good.... so fassst :zap: :muscle: :star_struck: 
![image](https://user-images.githubusercontent.com/5235127/224438557-79eefde1-5de2-4b5d-882f-195266e65053.png)

Thank you very much for your feedback @Mytherin :100:
